File: examples/python-stdio-demo/configure.py
#!/usr/bin/env python3
# Demo stdio command: read request, write one-line mutations via SDK.
import time
from lyenv_sdk import read_request, log, plugin_write_config, respond_ok

def main():
    req = read_request()
    # one-liner: write plugin local config
    plugin_write_config("hello.enabled", True, scope="plugin")
    # write global config
    plugin_write_config("demo.last_run_at", time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()), scope="global")
    log(f"os={req.get('system',{})}")
    respond_ok("ok")

if __name__ == "__main__":
    main()

File: examples/node-stdio-demo/configure.js
#!/usr/bin/env node
// Demo stdio command: read request, write one-line mutations via SDK.

const sdk = require("./lyenv_sdk.js");

async function main() {
  const req = await sdk.read_request();

  sdk.plugin_write_config("hello.enabled", true, "plugin");
  sdk.plugin_write_config("demo.last_run_at", new Date().toISOString(), "global");
  sdk.log(`os=${JSON.stringify(req.system || {})}`);

  sdk.respond_ok("ok");
}

main().catch((err) => {
  try {
    sdk.respond_error(err.message || String(err));
  } catch {
    process.stderr.write(String(err) + "\n");
    process.exit(1);
  }
});
